diff --git a/apriltags/src/TagDetection.cc b/apriltags/src/TagDetection.cc
index e25f4fc..99c3711 100644
--- a/apriltags/src/TagDetection.cc
+++ b/apriltags/src/TagDetection.cc
@@ -91,7 +91,7 @@ Eigen::Matrix4d TagDetection::getRelativeTransform(double tag_size, double fx, d
   imgPts.push_back(cv::Point2f(p3.first, p3.second));
   imgPts.push_back(cv::Point2f(p4.first, p4.second));
 
-  cv::Mat rvec, tvec;
+  cv::Matx31d rvec, tvec;
   cv::Matx33f cameraMatrix(
                            fx, 0, px,
                            0, fy, py,
@@ -105,7 +105,7 @@ Eigen::Matrix4d TagDetection::getRelativeTransform(double tag_size, double fx, d
 
   Eigen::Matrix4d T; 
   T.topLeftCorner(3,3) = wRo;
-  T.col(3).head(3) << tvec.at<double>(0), tvec.at<double>(1), tvec.at<double>(2);
+  T.col(3).head(3) << tvec(0), tvec(1), tvec(2);
   T.row(3) << 0,0,0,1;
 
   return T;
